### PR TITLE
Video Segemntation On Short Videos

### PR DESCRIPTION
**Pull request by lkiesow**
Our video segmentation works extremely bad on short videos, e.g. trying
to divide a 0.5 seconds video into 100 segments, causing the segments to
be shorter than a single frame which then caused problems with the
slide-preview extraction.

Assuming that a segmentation on extremely short videos is useless at
best and causes workflow failures or services in error states at worst,
this patch will make Opencast skip the segmentation on videos shorter
than 15 seconds.

This is related to #1219 and #1326

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] be against the correct branch (features can only go into develop)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated testing
* [ ] have a clean commit history
* [ ] have proper commit messages (title and body) for all commits
* [ ] have appropriate tags applied

### CODE REVIEW

**Comment by mtneug** [comment]
Maybe make this configurable? I can imagine that it doesn't make sense for even longer videos (1-5 min.).

**Comment by lkiesow** [comment]
Not doing that. It's nothing anyone will really configure ever if there is a sensible default.
I can increase this to maybe 30sec though?

**Comment by mtneug** [comment]
Fair enough, though I would have misused this safety measure to skip videos of up to 5 min. where I don't think segmentation would be beneficial 😄.

Just as a safety measure 30 sec. or even 15 sec. seems fine to me. What I don't know, however, is if we should somehow consider [`absoluteMin`](https://github.com/opencast/opencast/blob/f940467be0dc3fcb52bcfed6c7aa99f92bb5514d/etc/org.opencastproject.videosegmenter.ffmpeg.VideoSegmenterServiceImpl.cfg#L38). In theory, you could set it to a very high value such that any number of seconds at given fps would lead to an error.

File: modules/videosegmenter-workflowoperation/src/main/java/org/opencastproject/workflow/handler/videosegmenter/VideoSegmenterWorkflowOperationHandler.java [hunk 1]
```diff
@@ -110,6 +110,12 @@ public WorkflowOperationResult start(final WorkflowInstance workflowInstance, Jo
     }
     Track track = candidates.get(0);
 
+    // Skip operation if media is shorter than 15s since we won't generate a sensible segmentation on such a short
+    // video anyway.
+    if (track.getDuration() != null && track.getDuration() < 15000) {
```
Comment (gregorydlogan) [inline modules/videosegmenter-workflowoperation/src/main/java/org/opencastproject/workflow/handler/videosegmenter/VideoSegmenterWorkflowOperationHandler.java]: 15 seconds is fine to me, although I somewhat agree with Matthias in that sub 5 minute videos don't really benefit from segmentation.

My complaint here whoever, is somewhat simpler: Magic numbers.  IMO this should be a constant properly defined so it shows up in the javadocs.

**Review by gregorydlogan — CHANGES_REQUESTED** [review]

**Comment by lkiesow** [comment]
- added variable for time
- set time to 30sec

**Pull request merged**